service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceClient.java

```diff
@@ -85,11 +92,9 @@ protected ServiceClient(IotHubConnectionString iotHubConnectionString, IotHubSer
      */
     public void open() throws IOException
     {
-        // Codes_SRS_SERVICE_SDK_JAVA_SERVICECLIENT_12_008: [The function shall throw IOException if the member AMQP sender object has not been initialized]
-        if (this.amqpMessageSender == null)
-        {
-            throw new IOException("AMQP sender is not initialized");
-        }
+        this.sasToken = new IotHubServiceSasToken(iotHubConnectionString, sasTokenExpiryTime).toString();
```

Comment:
"tostring()" (lowercase) in "IotHubServiceSasToken.java" is implemented to return the instance member "token". This is misleading and conflicting with Java's "Object" class's  "toString() (lowerCamelCase) which returns the string representation of the Object itself. 
Suggest to rename this to "getSaSToken()" and provide an overridden toString() method to return custom string representation if/where needed.

Reply:
please ignore the part of the comment on lowercase tostring - not sure where I saw that.

Reply:
This actually looks good. Please ignore my previous comments :(